web_app/utils/zklend.py

```diff
@@ -0,0 +1,192 @@
+    "Withdrawal": "process_withdrawal_event",
+    "zklend::market::Market::Withdrawal": "process_withdrawal_event",
+    "Borrowing": "process_borrowing_event",
+    "zklend::market::Market::Borrowing": "process_borrowing_event",
+    "Repayment": "process_repayment_event",
+    "zklend::market::Market::Repayment": "process_repayment_event",
+    "Liquidation": "process_liquidation_event",
+    "zklend::market::Market::Liquidation": "process_liquidation_event",
+}
+
+
+class ZkLendLoanEntity(LoanEntity):
+    """
+    A class that describes the zkLend loan entity. On top of the abstract `LoanEntity`, it implements the `deposit` and
+    `collateral_enabled` attributes in order to help with accounting for the changes in collateral. This is because
+    under zkLend, collateral is the amount deposited that is specificaly flagged with `collateral_enabled` set to True
+    for the given token. To properly account for the changes in collateral, we must hold the information about the
+    given token's deposits being enabled as collateral or not and the amount of the deposits. We keep all balances in raw
+    amounts.
+    """
+
+    TOKEN_SETTINGS: dict[str, TokenSettings] = TOKEN_SETTINGS
+
+    def __init__(self) -> None:
+        super().__init__()
+        self.deposit: Portfolio = Portfolio()
+        self.collateral_enabled: TokenValues = TokenValues(init_value=False)
+
+    def compute_health_factor(
+        self,
+        standardized: bool,
+        collateral_interest_rate_models: InterestRateModels | None = None,
+        debt_interest_rate_models: InterestRateModels | None = None,
+        prices: TokenValues | None = TokenValues(),
+        risk_adjusted_collateral_usd: Decimal | None = None,
+        debt_usd: Decimal | None = None,
+    ) -> Decimal:
+        """
+        Compute's health ratio factor for zkLend loans.
+        :param standardized: InterestRateModels | None = None
+        :param collateral_interest_rate_models: InterestRateModels | None = None
+        :param debt_interest_rate_models: InterestRateModels | None = None
+        :param prices: TokenValues | None = None
+        :param risk_adjusted_collateral_usd: Decimal | None = None
+        :param debt_usd: Decimal | None = None
+        :return: Decimal
+        """
+        if risk_adjusted_collateral_usd is None:
+            risk_adjusted_collateral_usd = self.compute_collateral_usd(
+                collateral_interest_rate_models=collateral_interest_rate_models,
+                prices=prices,
+                risk_adjusted=True,
+            )
+
+        if debt_usd is None:
+            debt_usd = self.compute_debt_usd(
+                debt_interest_rate_models=debt_interest_rate_models,
+                prices=prices,
+                risk_adjusted=False,
+            )
+
+        if standardized:
+            denominator = debt_usd
+        else:
+            denominator = debt_usd
+
+        if denominator == Decimal("0"):
```

Comment:
Feel free to remove `denominator` and use just `debt_usd` as it's just extra step